launcher/src/main/java/org/apache/spark/launcher/AbstractCommandBuilder.java

```diff
@@ -181,9 +181,16 @@ void addOptionString(List<String> cmd, String options) {
         }
       }
       if (isTesting) {
+        boolean isGitHubActionsBuild = System.getenv("GITHUB_ACTIONS") != null;
```

Comment:
Is it inevitable? It looks a little weird to me to have `GITHUB_ACTIONS` in this part.

Reply:
It is weird .. I will take a fresh look tomorrow and see if there's a better way. To avoid such bandaid fix, we should basically make a fix like https://github.com/apache/spark/pull/12334 .. #12334 wasn't even a complete fix if I am not mistaken. At least I think It will need a big change ..